lua/effects/eff_term_goodarc.lua

```diff
@@ -0,0 +1,436 @@
+            self.Points[#self.Points + 1] = basePos + right * offsetX + up * offsetY
+
+        end
+    end
+
+    self.Points[#self.Points + 1] = endPos
+
+    -- Generate branches
+    if self.DoBranches and #self.Points >= 4 then
+        self:GenerateBranches( direction, length )
+
+    else
+        self.Branches = nil
+
+    end
+end
+
+function EFFECT:GenerateBranches( mainDir, mainLength )
+    self.Branches = {}
+
+    local branchCount = self.BranchCount or math.Clamp( math.floor( self.Segments / 2.5 ), 1, 6 )
+    local usedIndices = {}
+
+    for _ = 1, branchCount do
+        local attempts = 0
+        local pointIdx
+
+        repeat
+            pointIdx = math.random( 2, math.max( 2, #self.Points - 2 ) )
+            attempts = attempts + 1
+
+        until not usedIndices[pointIdx] or attempts > 10
+
+        usedIndices[pointIdx] = true
+
+        local branchStart = self.Points[pointIdx]
+        local branchDir = ( VectorRand() + mainDir * 0.2 ):GetNormalized()
+        local branchLen = mainLength * math.Rand( 0.15, 0.4 )
+        local branchSegs = math.random( 2, 4 )
+
+        local branch = {branchStart}
+        local bAng = branchDir:Angle()
+        local bRight, bUp = bAng:Right(), bAng:Up()
+
+        for j = 1, branchSegs do
+            local t = j / branchSegs
+            local pos = branchStart + branchDir * branchLen * t
+            local jitter = self.Jitter * 0.5 * ( 1 - t * 0.7 )
+            pos = pos + bRight * math.Rand( -1, 1 ) * jitter + bUp * math.Rand( -1, 1 ) * jitter
+            branch[#branch + 1] = pos
+
+        end
+
+        branch.width = math.Rand( 0.4, 0.7 )
+        self.Branches[#self.Branches + 1] = branch
+
+        -- Sub-branches
+        if branchSegs >= 3 and math.random() > 0.5 then
+            local subStart = branch[2]
+            local subDir = ( VectorRand() + branchDir * 0.1 ):GetNormalized()
+            local subLen = branchLen * math.Rand( 0.3, 0.5 )
+
+            local subBranch = {subStart}
+            for k = 1, 2 do
+                local t = k / 2
+                local pos = subStart + subDir * subLen * t
+                pos = pos + VectorRand() * self.Jitter * 0.3 * ( 1 - t )
+                subBranch[#subBranch + 1] = pos
+
+            end
+
+            subBranch.width = branch.width * 0.5
+            self.Branches[#self.Branches + 1] = subBranch
+
+        end
+    end
+end
+
+function EFFECT:Think()
+    if CurTime() >= self.DieTime then return false end
+
+    if self.NoFlicker then return true end
+
+    self.NextFlicker = self.NextFlicker or 0
+    if CurTime() >= self.NextFlicker then return true end
+
+    self:GenerateArc()
+    self.NextFlicker = CurTime() + math.Rand( 0.015, 0.035 ) / self.FlickerMul
+
+    return true
+
+end
+
+function EFFECT:Render()
+    local timeLeft = self.DieTime - CurTime()
+    if timeLeft <= 0 then return end
+
+    local alpha = self.NoFade and 1 or math.Clamp( timeLeft / self.Duration, 0, 1 )
```

Comment:
this variable should be renamed to fade, alpha implies it JUST makes it transparent, but it also makes it smaller